src/Uno.Extensions.Http.Kiota/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,132 @@
+	/// <param name="name">[Optional] The name for locating endpoint information in appsettings.</param>
+	/// <param name="configure">[Optional] A callback for configuring the endpoint.</param>
+	/// <returns>The updated <see cref="IServiceCollection"/> with the registered Kiota client.</returns>
+	public static IServiceCollection AddKiotaClient<TClient>(
+		this IServiceCollection services,
+		HostBuilderContext context,
+		EndpointOptions? options = null,
+		string? name = null,
+		Func<IHttpClientBuilder, EndpointOptions?, IHttpClientBuilder>? configure = null
+	)
+		where TClient : class =>
+		services.AddKiotaClientWithEndpoint<TClient, EndpointOptions>(context, options, name, configure);
+
+	/// <summary>
+	/// Registers a Kiota client with the specified <paramref name="name"/> and supports additional endpoint options.
+	/// </summary>
+	/// <typeparam name="TClient">The Kiota client type to register.</typeparam>
+	/// <typeparam name="TEndpoint">The type of endpoint to register.</typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to register the client with.</param>
+	/// <param name="context">The <see cref="HostBuilderContext"/> providing the hosting context.</param>
+	/// <param name="options">[Optional] The endpoint options for the client (loaded from appsettings if not specified).</param>
+	/// <param name="name">[Optional] The name for locating endpoint information in appsettings.</param>
+	/// <param name="configure">[Optional] A callback for configuring the endpoint.</param>
+	/// <returns>The updated <see cref="IServiceCollection"/> with the registered Kiota client.</returns>
+	public static IServiceCollection AddKiotaClientWithEndpoint<TClient, TEndpoint>(
+		this IServiceCollection services,
+		HostBuilderContext context,
+		TEndpoint? options = null,
+		string? name = null,
+		Func<IHttpClientBuilder, TEndpoint?, IHttpClientBuilder>? configure = null
+	)
+		where TClient : class
+		where TEndpoint : EndpointOptions, new()
+	{
+		services.AddKiotaHandlers();
+		services.AddSingleton<AllowedHostsValidator>(sp => new AllowedHostsValidator());
+
+		return services.AddClientWithEndpoint<TClient, TEndpoint>(
+				context,
+				options,
+				name: name ?? typeof(TClient).FullName ?? "DefaultClient",
+				httpClientFactory: (s, c) => s.AddHttpClient<TClient>(name ?? typeof(TClient).FullName ?? "DefaultClient")
+					.AttachKiotaHandlers()
+					.ConfigureHttpClient(client =>
+					{
+						if (options?.Url != null)
+						{
+							client.BaseAddress = new Uri(options.Url);
+						}
+					}),
+				configure: configure
+			)
+			.AddSingleton<IRequestAdapter, HttpClientRequestAdapter>(sp =>
+			{
+				var tokenCache = sp.GetRequiredService<ITokenCache>();
+				var allowedHostsValidator = sp.GetRequiredService<AllowedHostsValidator>();
+				var accessTokenProvider = new TokenProvider(tokenCache, allowedHostsValidator);
+				var authProvider = new BaseBearerTokenAuthenticationProvider(accessTokenProvider);
```

Comment:
I think we will need to integrate this with the Uno.Http.Authentication instead of using the auth logic from Kiota, right @nickrandolph?